strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/maven/MavenArtifactController.java

```diff
@@ -119,6 +123,76 @@ public void download(@ApiParam(value = "The storageId", required = true)
                          HttpServletRequest request,
                          HttpServletResponse response)
             throws Exception
+    {
+        ArtifactInputStream is = getArtifactInputStream(storageId,
+                                                        repositoryId,
+                                                        path,
+                                                        response,
+                                                        request,
+                                                        httpHeaders);
+        
+        if(is == null)
+        {
+            return;
+        }
+        RepositoryFileAttributes fileAttributes = getRepositoryFileAttributes(storageId,
+                                                                              repositoryId,
+                                                                              path,
+                                                                              response);
+
+        copyToResponse(is, response);
+        
+        setHeaders(is, response, fileAttributes, path);
+
+        logger.debug("Download succeeded.");
+    }
+    
+    @ApiOperation(value = "Used to retrieve headers of an artifact", position = 2)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = ""),
+                            @ApiResponse(code = 400, message = "An error occurred.") })
+    @PreAuthorize("hasAuthority('ARTIFACTS_RESOLVE')")
+    @RequestMapping(value = { "{storageId}/{repositoryId}/{path:.+}" }, method = RequestMethod.HEAD)
+    public void getHeaders(@ApiParam(value = "The storageId", required = true)
+                           @PathVariable String storageId,
+                           @ApiParam(value = "The repositoryId", required = true)
+                           @PathVariable String repositoryId,
+                           @RequestHeader HttpHeaders httpHeaders,
+                           @PathVariable String path,
+                           HttpServletRequest request,
+                           HttpServletResponse response)
+            throws Exception
+    {
+       ArtifactInputStream is = getArtifactInputStream(storageId,
+                                                       repositoryId,
+                                                       path,
+                                                       response,
+                                                       request,
+                                                       httpHeaders);
+       
+       if(is == null)
+       {
+           return;
+       }
+       
+       RepositoryFileAttributes fileAttributes = getRepositoryFileAttributes(storageId,
```

Comment:
`getRepositoryFileAttributes` may return `null` as far as I see. What then ? You don't handle it, or I am missing something ?

Reply:
I had not better idea about guessing the size. I mean... once you're done writing to the stream, you'll have the final number of bytes.

Do you know of a better way?

Reply:
We have 

`org.carlspring.strongbox.domain.ArtifactEntry#sizeInBytes` or `org.carlspring.strongbox.providers.io.RepositoryFileAttributes#size`